src/main/java/tracko/logic/parser/ParserUtil.java

```diff
@@ -121,4 +124,47 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
         }
         return tagSet;
     }
+
+    /**
+     * Parses a {@code String itemName} into a {@code ItemName}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code itemName} is invalid.
+     */
+    public static ItemName parseItemName(String itemName) throws ParseException {
+        requireNonNull(itemName);
+        String trimmedName = itemName.trim();
+        if (!ItemName.isValidItemName(trimmedName)) {
+            throw new ParseException(ItemName.MESSAGE_CONSTRAINTS);
+        }
+        return new ItemName(trimmedName);
+    }
+
+    /**
+     * Parses {@code String quantity} into an {@code Quantity} and returns it. Leading and trailing whitespaces will be
+     * trimmed.
+     * @throws ParseException if the specified quantity is invalid (not non-zero unsigned integer).
+     */
+    public static Quantity parseQuantity(String quantity) throws ParseException {
+        String trimmedQuantity = quantity.trim();
```

Comment:
Also wanted to point this part out; I think you missed the `requireNonNull(quantity)` here!

Reply:
Oh woops I meant the line above this one haha :")

Reply:
Yep, `requireNonNull` is included in the `isNonNegativeUnsignedInteger` method!